scripts/scil_denoise_sf.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Script to compute per-vertices hemisphere-aware (asymmetric) Gaussian
+filtering of spherical functions (SF) given an array of spherical harmonics
+(SH) coefficients.
+
+The resulting SF can be expressed using a full SH basis (to keep the
+asymmetry resulting from the filtering) or a symmetric SH basis (where the
+effect of the filtering is a simple denoising).
+"""
+
+import argparse
+import logging
+import os
+
+import nibabel as nib
+import numpy as np
+
+from dipy.data import SPHERE_FILES
+
+from scilpy.io.utils import (add_overwrite_arg,
+                             assert_inputs_exist,
+                             add_sh_basis_args,
+                             assert_outputs_exist)
+
+from scilpy.denoise.asym_enhancement import local_asym_gaussian_filtering
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_sh',
+                   help='Path to the input file.')
+
+    p.add_argument('out_sh',
+                   help='File name for averaged signal.')
+
+    p.add_argument('--out_mask', default='mask.nii.gz',
+                   help='File name for output mask. [%(default)s]')
```

Comment:
What is in this mask?

Reply:
The mask contains voxels where the input signal is greater than 0 (or some `epsilon`), it's been useful to me for comparing the input/output signal only in voxels where there was an input signal in the first place.

Come to think of it, maybe it isn't worthy of an output in this script... Thoughts?

Reply:
I removed the mask output.